### ISSUE TITLE
fp16 training

### ISSUE THREAD

**Issue by tdomhan** [microsoft/MASS#38]
Turning on the `--fp16 True` flag I receive the following error
```
    assert len(modules) == 1, "fp16 not implemented for more than one module"
AssertionError: fp16 not implemented for more than one module
```

Any idea why this could be? Is fp16 supported?

**Comment by StillKeepTry** [comment]
This function is restricted by the old API in apex. I will use the latest apex API to solve this issue later.

**Comment by tdomhan** [comment]
oh ok. Thanks for the fast reply. So using an older version of apex would solve the issue? Which version/commit do you recommend?

**Comment by tdomhan** [comment]
I realized you are probably referring to https://nvidia.github.io/apex/amp.html?highlight=fp16_optimizer So this means fp16 can not be used unless you rewrite the MASS code, no matter which apex version is used.

**Comment by StillKeepTry** [comment]
@tdomhan Yes, I will rewrite the fp16 code in our project to support multiple modules.

**Comment by tdomhan** [comment]
great, thank you!

**Comment by gabrielhuang** [comment]
Hi all,

Has this been implemented ? I am still getting the "more than one module" error.

Thanks !
